src/censure.jl

```diff
@@ -0,0 +1,247 @@
+    	in_sum += A + D - B - C
+
+    	topleft = I + CartesianIndex(- m_out2 - 1, - m_out2 - OF.n_out - 1)
+    	topright = I + CartesianIndex(m_out2, - m_out2 - OF.n_out - 1)
+    	bottomleft = I + CartesianIndex(- m_out2 - 1, m_out2 + OF.n_out)
+    	bottomright = I + CartesianIndex(m_out2, m_out2 + OF.n_out)
+    	A = checkbounds(Bool, int_img, topleft) ? int_img[topleft] : 0.0
+    	B = checkbounds(Bool, int_img, topright) ? int_img[topright] : 0.0
+    	C = checkbounds(Bool, int_img, bottomleft) ? int_img[bottomleft] : 0.0
+    	D = checkbounds(Bool, int_img, bottomright) ? int_img[bottomright] : 0.0
+    	out_sum = A + D - B - C
+
+    	trap_top_right = bottomright
+    	trap_bot_right = I + CartesianIndex(m_out2 + OF.n_out, m_out2)
+    	trap_top_left = I + CartesianIndex(m_out2, - m_out2 - OF.n_out)
+    	trap_bot_left = I + CartesianIndex(m_out2 + OF.n_out, - m_out2 - 1)
+    	A = checkbounds(Bool, ls_img, trap_top_left) ? ls_img[trap_top_left] : 0.0
+    	B = checkbounds(Bool, rs_img, trap_top_right) ? rs_img[trap_top_right] : 0.0
+    	C = checkbounds(Bool, ls_img, trap_bot_left) ? ls_img[trap_bot_left] : 0.0
+    	D = checkbounds(Bool, rs_img, trap_bot_right) ? rs_img[trap_bot_right] : 0.0
+    	out_sum += A + D - B - C
+
+    	trap_top_right = I + CartesianIndex(- m_out2 - OF.n_out - 1, m_out2 - 1)
+    	trap_top_left = I + CartesianIndex(- m_out2 - OF.n_out - 1, - m_out2)
+    	trap_bot_right = I + CartesianIndex(- m_out2 - 1, m_out2 + OF.n_out - 1)
+    	trap_bot_left = I + CartesianIndex(- m_out2 - 1, - m_out2 - OF.n_out)
+    	A = checkbounds(Bool, rs_img, trap_top_left) ? rs_img[trap_top_left] : 0.0
+    	B = checkbounds(Bool, ls_img, trap_top_right) ? ls_img[trap_top_right] : 0.0
+    	C = checkbounds(Bool, rs_img, trap_bot_left) ? rs_img[trap_bot_left] : 0.0
+    	D = checkbounds(Bool, ls_img, trap_bot_right) ? ls_img[trap_bot_right] : 0.0
+    	out_sum += A + D - B - C
+    	out_sum = out_sum - in_sum
+
+    	response[I] = in_sum * filter.in_weight - out_sum * filer.out_weight
+    end
+	response
+	
+end
+
+function CENSURE(; smallest::Integer = 1, largest::Integer = 7, filter::Type = BoxFilter, response_threshold::Number = 0.15, line_threshold::Number = 10)
+    CENSURE{filter}(smallest, largest, filter, _get_filter_stack(filter, smallest, largest), response_threshold, line_threshold)
+end
+
+function censure{T, F}(img::AbstractArray{T, 2}, params::CENSURE{F})
+	int_img = _get_integral_image(img, params.filter_stack[1])
+	responses = map(f -> _filter_response(int_img, f), params.filter_stack)
+    response_matrix = reshape(hcat(responses...), size(img)..., size(responses)...)
+    minima, maxima = extrema_filter(convert(Array{Float64}, padarray(response_matrix, [1, 1, 1], [1, 1, 1], "replicate")), [3, 3, 3])
+    features = map(i -> (minima[i] == response_matrix[i] || maxima[i] == response_matrix[i]) && ( response_matrix[i] > params.response_threshold ), CartesianRange(size(response_matrix)))
```

Comment:
What's the array size? (Looks like it's an `Array{64,3}`, right?)

Looks like `extrema_filter` allocates temporary storage frequently, but I haven't looked at how it works all that carefully. It might be possible to speed it up by reusing the same buffer many times.